--- node_modules/@angular/cli/models/webpack-configs/production.js	2017-05-24 15:36:43.000000000 -0700
+++ node_modules/@angular/cli/models/webpack-configs/production.js	2017-05-24 15:37:04.000000000 -0700
@@ -85,6 +85,7 @@
                 'NODE_ENV': 'production'
             }),
             new webpack.HashedModuleIdsPlugin(),
+            new webpack.optimize.ModuleConcatenationPlugin(),
             new (require("ngo").PurifyPlugin)(),
             new webpack.optimize.UglifyJsPlugin({
                 mangle: true,
