--- node_modules/@angular/cli/models/webpack-configs/browser.js	2017-11-01 12:07:56.000000000 +0100
+++ node_modules/@angular/cli/models/webpack-configs/browser.js	2017-11-01 12:08:12.000000000 +0100
@@ -104,11 +104,11 @@
         ].concat(extraPlugins),
         node: {
             fs: 'empty',
-            global: true,
+            global: false,
             crypto: 'empty',
             tls: 'empty',
             net: 'empty',
-            process: true,
+            process: false,
             module: false,
             clearImmediate: false,
             setImmediate: false
