diff --git a/src/utils/version.js b/src/utils/version.js
index 300e98cbde4c2e0e6892b44cb8db22df96692227..ee65c768b4711def71e9f028b707080ac3ed2a7b 100755
--- a/src/utils/version.js
+++ b/src/utils/version.js
@@ -34,10 +34,10 @@ function assertCompatibleAngularVersion(projectRoot) {
         return;
     }
     const angularVersion = new semver_1.SemVer(angularPkgJson['version']);
-    if (!(0, semver_1.satisfies)(angularVersion, supportedAngularSemver, { includePrerelease: true })) {
-        console.error(`Error: The current version of "@angular/build" supports Angular versions ${supportedAngularSemver},\n` +
-            `but detected Angular version ${angularVersion} instead.\n` +
-            'Please visit the link below to find instructions on how to update Angular.\nhttps://update.angular.dev/');
-        process.exit(3);
-    }
+    // if (!(0, semver_1.satisfies)(angularVersion, supportedAngularSemver, { includePrerelease: true })) {
+    //     console.error(`Error: The current version of "@angular/build" supports Angular versions ${supportedAngularSemver},\n` +
+    //         `but detected Angular version ${angularVersion} instead.\n` +
+    //         'Please visit the link below to find instructions on how to update Angular.\nhttps://update.angular.dev/');
+    //     process.exit(3);
+    // }
 }
