diff --git a/fesm2022/ssr.mjs b/fesm2022/ssr.mjs
index 4976c181079b86773efbd49f5101fb1cdd670205..b20e3d886663011e31ba15a40e72969b92e39fd2 100755
--- a/fesm2022/ssr.mjs
+++ b/fesm2022/ssr.mjs
@@ -352,7 +352,7 @@ async function renderAngular(html, bootstrap, url, platformProviders, serverCont
             applicationRef = moduleRef.injector.get(ApplicationRef);
         }
         else {
-            applicationRef = await bootstrap();
+            applicationRef = await bootstrap({platformRef});
         }
         // Block until application is stable.
         await applicationRef.whenStable();
@@ -1244,7 +1244,7 @@ async function getRoutesFromAngularRouterConfig(bootstrap, document, url, invoke
             applicationRef = moduleRef.injector.get(ApplicationRef);
         }
         else {
-            applicationRef = await bootstrap();
+            applicationRef = await bootstrap({platformRef});
         }
         const injector = applicationRef.injector;
         const router = injector.get(Router);
