diff --git a/node_modules/@angular/ng-dev/bundles/chunk-UNUYI25Y.mjs b/node_modules/@angular/ng-dev/bundles/chunk-UNUYI25Y.mjs
index ff6ab3e..ae4b291 100755
--- a/node_modules/@angular/ng-dev/bundles/chunk-UNUYI25Y.mjs
+++ b/node_modules/@angular/ng-dev/bundles/chunk-UNUYI25Y.mjs
@@ -820,7 +820,8 @@ function getCachedConfig() {
 }
 
 // bazel-out/k8-fastbuild/bin/ng-dev/utils/config.js
-var CONFIG_FILE_PATH = ".ng-dev/config.mjs";
+const [major, minor, patch] = process.versions.node.split('.').map(Number)
+var CONFIG_FILE_PATH = `.ng-dev/config.${major === 18 && minor < 19 ? 'mjs' : 'mts'}`;
 var USER_CONFIG_FILE_PATH = ".ng-dev.user";
 var userConfig = null;
 var setConfig = setCachedConfig;
