diff --git a/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl b/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
index 6cee158..815a55d 100755
--- a/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
@@ -1,5 +1,5 @@
 load("@npm//@angular/build-tooling/bazel:extract_types.bzl", "extract_types")
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary", "nodejs_test")

 nodejs_test_args = [
     # Needed so that node doesn't walk back to the source directory.
diff --git a/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel b/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
index 1c27a95..c239203 100755
--- a/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary")

 package(default_visibility = ["//visibility:public"])

diff --git a/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl b/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
index d6249b8..707ae39 100755
--- a/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
@@ -4,11 +4,11 @@
 # found in the LICENSE file at https://angular.io/license

 load("@build_bazel_rules_nodejs//:index.bzl", "npm_package_bin")
-load("@npm//@bazel/terser:index.bzl", "terser_minified")
-load("@npm//prettier:index.bzl", "prettier")
-load("@npm//@angular/build-tooling/bazel/esbuild:index.bzl", "esbuild", "esbuild_config")
 load("@npm//@angular/build-tooling/bazel:expand_template.bzl", "expand_template")
 load("@npm//@angular/build-tooling/bazel:filter_outputs.bzl", "filter_outputs")
+load("@npm//@angular/build-tooling/bazel/esbuild:index.bzl", "esbuild", "esbuild_config")
+load("@npm//@bazel/terser:index.bzl", "terser_minified")
+load("@npm//prettier:index.bzl", "prettier")

 def _create_esbuild_minify_options(debug = False):
     # The minify options match with the configuration used by the CLI. The whitespace
@@ -64,7 +64,8 @@ def app_bundle(
         name = "%s_esbuild_config" % name,
         config_file = ":%s_config_file" % name,
         deps = [
-            "@npm//@angular/compiler-cli",
+            "@angular//packages/compiler:npm_package",
+            "@angular//packages/compiler-cli:npm_package",
             "@npm//@angular/build-tooling/shared-scripts/angular-optimization:js_lib",
         ],
         **common_base_attributes
diff --git a/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel b/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
index 3a85025..12a2c55 100755
--- a/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary")
 load("@npm//@angular/build-tooling/bazel:defaults.bzl", "ts_library")

 package(default_visibility = ["//visibility:public"])
diff --git a/node_modules/@angular/build-tooling/bazel/integration/index.bzl b/node_modules/@angular/build-tooling/bazel/integration/index.bzl
index 6f99d65..1bd5183 100755
--- a/node_modules/@angular/build-tooling/bazel/integration/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/integration/index.bzl
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_test")

 def _serialize_file(file):
     """Serializes a file into a struct that matches the `BazelFileInfo` type in the
diff --git a/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl b/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
index af0f305..ce46302 100755
--- a/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
@@ -3,7 +3,7 @@
 # Use of this source code is governed by an MIT-style license that can be
 # found in the LICENSE file at https://angular.io/license

-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary", "nodejs_test")

 nodejs_args = ["--nobazel_run_linker"]

diff --git a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
index b2729ad..80ab926 100755
--- a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
@@ -24,7 +24,8 @@ js_library(
     srcs = [":js_lib_files"],
     deps = [
         "@npm//@angular/build",
-        "@npm//@angular/compiler-cli",
+        "@angular//packages/compiler:npm_package",
+        "@angular//packages/compiler-cli:npm_package",
         "@npm//@babel/core",
         "@npm//@babel/plugin-proposal-async-generator-functions",
     ],
