diff --git a/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl b/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
index 6cee158..815a55d 100755
--- a/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/api-golden/index.bzl
@@ -1,5 +1,5 @@
 load("@npm//@angular/build-tooling/bazel:extract_types.bzl", "extract_types")
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary", "nodejs_test")
 
 nodejs_test_args = [
     # Needed so that node doesn't walk back to the source directory.
diff --git a/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel b/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
index 1c27a95..c239203 100755
--- a/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/bazel/app-bundling/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary")
 
 package(default_visibility = ["//visibility:public"])
 
diff --git a/node_modules/@angular/build-tooling/bazel/app-bundling/esbuild.config-tmpl.mjs b/node_modules/@angular/build-tooling/bazel/app-bundling/esbuild.config-tmpl.mjs
index e4410ba..33d75cd 100755
--- a/node_modules/@angular/build-tooling/bazel/app-bundling/esbuild.config-tmpl.mjs
+++ b/node_modules/@angular/build-tooling/bazel/app-bundling/esbuild.config-tmpl.mjs
@@ -9,7 +9,7 @@
 import * as path from 'path';
 
 import {createEsbuildAngularOptimizePlugin} from '@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs';
-import {GLOBAL_DEFS_FOR_TERSER_WITH_AOT} from '@angular/compiler-cli/private/tooling';
+import {GLOBAL_DEFS_FOR_TERSER_WITH_AOT} from '@angular/compiler-cli/private/tooling.mjs';
 
 /** Root path pointing to the app bundle source entry-point file. */
 const entryPointSourceRootPath = path.normalize(`TMPL_ENTRY_POINT_ROOTPATH`);
diff --git a/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl b/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
index 1edd96d..3d423f4 100755
--- a/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/app-bundling/index.bzl
@@ -64,7 +64,7 @@ def app_bundle(
         name = "%s_esbuild_config" % name,
         config_file = ":%s_config_file" % name,
         deps = [
-            "@npm//@angular/compiler-cli",
+            "@angular//packages/compiler-cli/private",
             "@npm//@angular/build-tooling/shared-scripts/angular-optimization:js_lib",
         ],
         **common_base_attributes
diff --git a/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/benchmark_test.bzl b/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/benchmark_test.bzl
index d36afe6..fbdc9bb 100755
--- a/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/benchmark_test.bzl
+++ b/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/benchmark_test.bzl
@@ -1,4 +1,4 @@
-load("@npm//@bazel/protractor:index.bzl", "protractor_web_test_suite")
+load("@angular//tools:defaults.bzl", "protractor_web_test_suite")
 
 """
   Macro that can be used to define a benchmark test. This differentiates from
@@ -7,7 +7,7 @@ load("@npm//@bazel/protractor:index.bzl", "protractor_web_test_suite")
   unless explicitly requested.
 """
 
-def benchmark_test(name, server, tags = [], **kwargs):
+def benchmark_test(name, server, tags = [], data = [], **kwargs):
     protractor_web_test_suite(
         name = name,
         browsers = ["@npm//@angular/build-tooling/bazel/browsers/chromium:chromium"],
@@ -19,6 +19,7 @@ def benchmark_test(name, server, tags = [], **kwargs):
             "manual",
             "no-remote-exec",
         ],
+        data = data + ["@angular//packages/benchpress"],
         test_suite_tags = [
             "manual",
             "no-remote-exec",
diff --git a/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/component_benchmark.bzl b/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/component_benchmark.bzl
index 6213703..4d91aa3 100755
--- a/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/component_benchmark.bzl
+++ b/node_modules/@angular/build-tooling/bazel/benchmark/component_benchmark/component_benchmark.bzl
@@ -1,7 +1,7 @@
 load("@npm//@angular/build-tooling/bazel/app-bundling:index.bzl", "app_bundle")
 load("@npm//@angular/build-tooling/bazel/http-server:index.bzl", "http_server")
 load("@npm//@angular/build-tooling/bazel:expand_template.bzl", "expand_template")
-load("@npm//@angular/bazel:index.bzl", "ng_module")
+load("@angular//tools:defaults.bzl", "ng_module")
 load("@npm//@bazel/concatjs:index.bzl", "ts_library")
 load(":benchmark_test.bzl", "benchmark_test")
 
@@ -37,8 +37,8 @@ def component_benchmark(
         driver_deps,
         ng_srcs,
         ng_deps = [
-            "@npm//@angular/core",
-            "@npm//@angular/platform-browser",
+            "@angular//packages/core",
+            "@angular//packages/platform-browser",
         ],
         ng_assets = [],
         assets = None,
@@ -108,7 +108,7 @@ def component_benchmark(
         # checked by TypeScript. We add the dependency only for bundling to reduce the compilation
         # scope and to make it easier to replace this dependency inside the `angular/angular`
         # repository with its corresponding source target that does not come with any typings.
-        ng_bundle_deps.append("@npm//zone.js")
+        ng_bundle_deps.append("@angular//packages/zone.js/dist:zone")
 
     if not assets:
         html = prefix + "index.html"
diff --git a/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel b/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
index 04cc755..ec01017 100755
--- a/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/bazel/http-server/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary")
 load("@npm//@bazel/concatjs:index.bzl", "ts_library")
 
 package(default_visibility = ["//visibility:public"])
diff --git a/node_modules/@angular/build-tooling/bazel/integration/index.bzl b/node_modules/@angular/build-tooling/bazel/integration/index.bzl
index 662272d..9fc6330 100755
--- a/node_modules/@angular/build-tooling/bazel/integration/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/integration/index.bzl
@@ -1,4 +1,4 @@
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_test")
 
 def _serialize_file(file):
     """Serializes a file into a struct that matches the `BazelFileInfo` type in the
diff --git a/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl b/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
index 47f54aa..0fe449c 100755
--- a/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
+++ b/node_modules/@angular/build-tooling/bazel/map-size-tracking/index.bzl
@@ -3,7 +3,7 @@
 # Use of this source code is governed by an MIT-style license that can be
 # found in the LICENSE file at https://angular.io/license
 
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary", "nodejs_test")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary", "nodejs_test")
 
 nodejs_args = ["--nobazel_run_linker"]
 
diff --git a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
index 79afcb8..ebbb37b 100755
--- a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
+++ b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/BUILD.bazel
@@ -43,7 +43,8 @@ js_library(
     srcs = [":js_lib_files"],
     deps = [
         ":angular_devkit_plugins",
-        "@npm//@angular/compiler-cli",
+        "@angular//packages/compiler-cli",
+        "@angular//packages/compiler-cli/linker/babel",
         "@npm//@babel/core",
         "@npm//@babel/plugin-proposal-async-generator-functions",
     ],
diff --git a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
index f5908b9..d5fac4f 100755
--- a/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
+++ b/node_modules/@angular/build-tooling/shared-scripts/angular-optimization/esbuild-plugin.mjs
@@ -54,8 +54,8 @@ export async function createEsbuildAngularOptimizePlugin(opts, additionalBabelPl
 
   if (opts.enableLinker) {
     linkerCreator = {
-      compiler: await import('@angular/compiler-cli'),
-      babel: await import('@angular/compiler-cli/linker/babel'),
+      compiler: await import('@angular/compiler-cli/index.mjs'),
+      babel: await import('@angular/compiler-cli/linker/babel/index.mjs'),
     };
   }
 
