diff --git a/node_modules/@bazel/protractor/protractor_web_test.bzl b/node_modules/@bazel/protractor/protractor_web_test.bzl
index 77fcd99..8c65658 100755
--- a/node_modules/@bazel/protractor/protractor_web_test.bzl
+++ b/node_modules/@bazel/protractor/protractor_web_test.bzl
@@ -14,7 +14,7 @@
 "Run end-to-end tests with Protractor"
 
 load("@rules_nodejs//nodejs:providers.bzl", "JSModuleInfo")
-load("@build_bazel_rules_nodejs//:index.bzl", "nodejs_binary")
+load("@angular//tools/esm-interop:index.bzl", "nodejs_binary")
 load("@build_bazel_rules_nodejs//:providers.bzl", "ExternalNpmPackageInfo", "JSNamedModuleInfo", "node_modules_aspect")
 load("@build_bazel_rules_nodejs//internal/common:windows_utils.bzl", "create_windows_native_launcher_script", "is_windows")
 load("@io_bazel_rules_webtesting//web:web.bzl", "web_test_suite")
