diff --git nodejs/private/toolchains_repo.bzl nodejs/private/toolchains_repo.bzl
index 9dcdc60..4927e75 100644
--- nodejs/private/toolchains_repo.bzl
+++ nodejs/private/toolchains_repo.bzl
@@ -70,7 +70,7 @@ def _toolchains_repo_impl(repository_ctx):
 
 # Forward all the providers
 def _resolved_toolchain_impl(ctx):
-    toolchain_info = ctx.toolchains["@rules_nodejs//nodejs:toolchain_type"]
+    toolchain_info = ctx.toolchains["@build_bazel_rules_nodejs//nodejs:toolchain_type"]
     return [
         toolchain_info,
         toolchain_info.default,
@@ -82,7 +82,7 @@ def _resolved_toolchain_impl(ctx):
 # https://cs.opensource.google/bazel/bazel/+/master:tools/jdk/java_toolchain_alias.bzl
 resolved_toolchain = rule(
     implementation = _resolved_toolchain_impl,
-    toolchains = ["@rules_nodejs//nodejs:toolchain_type"],
+    toolchains = ["@build_bazel_rules_nodejs//nodejs:toolchain_type"],
     incompatible_use_toolchain_transition = True,
 )
 """
@@ -107,7 +107,7 @@ toolchain(
     exec_compatible_with = {compatible_with},
     target_compatible_with = {compatible_with},
     toolchain = "@{user_node_repository_name}_{platform}//:node_toolchain",
-    toolchain_type = "@rules_nodejs//nodejs:toolchain_type",
+    toolchain_type = "@build_bazel_rules_nodejs//nodejs:toolchain_type",
 )
 """.format(
             platform = platform,
diff --git nodejs/repositories.bzl nodejs/repositories.bzl
index 494217b..6dfb8fb 100644
--- nodejs/repositories.bzl
+++ nodejs/repositories.bzl
@@ -690,7 +690,7 @@ filegroup(
     # the platform attribute is only set when used from this file, not from build_bazel_rules_nodejs
     if repository_ctx.attr.platform:
         build_content += """
-load("@rules_nodejs//nodejs:toolchain.bzl", "node_toolchain")
+load("@build_bazel_rules_nodejs//nodejs:toolchain.bzl", "node_toolchain")
 node_toolchain(name = "node_toolchain", target_tool = ":node_bin")
 """
     repository_ctx.file("BUILD.bazel", content = build_content)
