diff --git sass/sass_wrapper.js sass/sass_wrapper.js
index ad1f7a6..957d56e 100644
--- sass/sass_wrapper.js
+++ sass/sass_wrapper.js
@@ -17,7 +17,17 @@ const fs = require('fs');
 const args = process.argv.slice(2);
 if (runAsWorker(args)) {
   debug('Starting Sass compiler persistent worker...');
-  runWorkerLoop(args => sass.cli_pkg_main_0_(args));
+  runWorkerLoop(args => {
+    return new Promise(resolve => {
+      const result = sass.cli_pkg_main_0_(args);
+      const originalThen = result.then$1$2$onError;
+      let dynamicType;
+      result.then$1$2$onError = (receiver, done, error, type) => dynamicType = type;
+      result.then$1();
+      result.then$1$2$onError = originalThen;
+      result.then$1$2$onError(null, () => resolve(true), () => resolve(false), dynamicType);
+    });
+  });
   // Note: intentionally don't process.exit() here, because runWorkerLoop
   // is waiting for async callbacks from node.
 } else {
